src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1724,26 +1724,32 @@ abstract class RefChecks extends Transform {
       def applyDepth: Int = {
         def loop(t: Tree, d: Int): Int =
           t match {
-            case Apply(f, _) => loop(f, d+1)
+            case Apply(t, _) => loop(t, d+1)
             case _ => d
           }
         loop(fn, 0)
       }
-      def isAssertParadigm(params: List[Symbol]): Boolean = !sym.isConstructor && !sym.isCaseApplyOrUnapply && {
+      def isAssertParadigm(params: List[Symbol]): Boolean =
         params match {
           case h :: t => h.tpe == BooleanTpe && !t.exists(_.tpe == BooleanTpe)
           case _ => false
         }
-      }
-      if (settings.lintNamedBooleans && !sym.isJavaDefined && !args.isEmpty) {
+      if (settings.warnNamedLiteral.value && !sym.isJavaDefined && !args.isEmpty) {
         val params = sym.paramLists(applyDepth)
-        if (!isAssertParadigm(params))
-          foreach2(args, params)((arg, param) => arg match {
-            case Literal(Constant(_: Boolean))
-            if arg.hasAttachment[UnnamedArg.type] && param.tpe.typeSymbol == BooleanClass && !param.deprecatedParamName.contains(nme.NO_NAME) =>
-              runReporting.warning(arg.pos, s"Boolean literals should be passed using named argument syntax for parameter ${param.name}.", WarningCategory.LintNamedBooleans, sym)
-            case _ =>
-          })
+        if (!settings.Ylint.value || !isAssertParadigm(params)) {
+          val suspicious = args.lazyZip(params).iterator
+            .filter {
+              case (arg @ Literal(Constant(_: Boolean)) , param) =>
+                arg.hasAttachment[UnnamedArg.type] && param.tpe.typeSymbol == BooleanClass && !param.deprecatedParamName.contains(nme.NO_NAME)
+              case _ => false
+            }
+            .toList
+          val warn = if (settings.Ylint.value) suspicious.nonEmpty else suspicious.lengthCompare(2) >= 0
```

Comment:
Maybe we should also warn if there's one literal arg but the function has multiple boolean parameters

```scala
scala> def f(x: Int, y: Boolean = true, z: Boolean = true) = 0
def f(x: Int, y: Boolean, z: Boolean): Int

scala> f(1, y = true, false) // warn?
val res12: Int = 0

scala> f(1, true) // warn?
val res13: Int = 0
```

Otherwise warning only for multiple literals seems a good way to tone it down.

Reply:
Somewhere I commented that I thought what the user wrote at use site is more relevant than the definition. That is an untested claim.

Reply:
The paradigm here is to consider "effectively unnamed boolean args", which are unnamed literals and default args. The relaxed check warns if there are at least 2 effectively unnamed args, at least one of which is a literal.